### PR TITLE
Filter out empty artifact profile blobs

### PR DESCRIPTION
**Pull request by hglkrijger**
```
2018/06/22 07:16:03.677467 ERROR ExtHandler Event: name=WALinuxAgent, op=ArtifactsProfileBlob, message=Content: [<0x00><0x00><0x00><0x00>
```
- add a check for null only content when the blob has not been written yet
- only log telemetry for failures when `log_event` is true
- add unit test for empty strings

### CODE REVIEW

File: azurelinuxagent/common/event.py [hunk 1]
```diff
@@ -256,7 +256,7 @@ def add_event(self,
                   is_internal=False,
                   log_event=True):
 
-        if not is_success or log_event:
+        if not is_success and log_event:
```
Comment (jasonzio) [inline azurelinuxagent/common/event.py]: Consider adding parentheses to make precedence clear without requiring the reader to look up whether `not` or `and` binds more tightly.

File: tests/utils/test_text_util.py [hunk 2]
```diff
@@ -159,6 +159,33 @@ def test_hash_list(self):
...
+
+        self.assertFalse(textutil.is_str_empty(u' \x01 '))
+        self.assertFalse(textutil.is_str_empty(u'foo'))
+        self.assertFalse(textutil.is_str_empty('bar'))
+
+        self.assertFalse(textutil.is_str_none_or_whitespace(u'\x00'))
+        self.assertFalse(textutil.is_str_none_or_whitespace(u' \x00 '))
+
+        self.assertTrue(textutil.is_str_empty(u'\x00'))
+        self.assertTrue(textutil.is_str_empty(u' \x00 '))
```
Comment (jasonzio) [inline tests/utils/test_text_util.py]: Consider pulling these strings into named variables (which can live within this method) so you can be certain you're invoking the two methods with identical strings.

File: tests/utils/test_text_util.py [hunk 3]
```diff
@@ -159,6 +159,33 @@ def test_hash_list(self):
         self.assertEqual(result_from_list, hash_from_string.digest())
         self.assertEqual(hash_from_string.hexdigest(), '6367c48dd193d56ea7b0baad25b19455e529f5ee')
 
+    def test_empty_strings(self):
+        self.assertTrue(textutil.is_str_none_or_whitespace(' '))
```
Comment (boumenot) [inline tests/utils/test_text_util.py]: textutil.is_str_none_or_whitespace(None) ??

File: tests/utils/test_text_util.py (hunk 2 above)
Comment (jasonzio) [inline tests/utils/test_text_util.py]: These last two cases are critical: they identify scenarios where the two methods are _required_ to return different values. Consider reframing those test cases to assert that the distinction exists. That is...
```
hex_null = u'\x00'
hex_null_with_blanks = u' \x00 '
self.assertTrue(textutil.is_str_empty(hex_null) and not textutil.is_str_none_or_whitespace(hex_null))
self.assertTrue(textutil.is_str_empty(hex_null_with_blanks) and not textutil.is_str_none_or_whitespace(hex_null_with_blanks))
```

File: tests/utils/test_text_util.py (hunk 3 above)
Comment (hglkrijger) [reply tests/utils/test_text_util.py]: Updated

File: azurelinuxagent/common/event.py (hunk 1 above)
Comment (hglkrijger) [reply azurelinuxagent/common/event.py]: Updated

File: tests/utils/test_text_util.py (hunk 2 above)
Comment (hglkrijger) [reply tests/utils/test_text_util.py]: Updated

Comment (hglkrijger) [reply tests/utils/test_text_util.py]: Updated
